iotdb/src/main/java/org/apache/iotdb/db/query/reader/sequence/SealedTsFilesReader.java

```diff
@@ -90,22 +91,22 @@ public boolean hasNext() throws IOException {
           hasCachedData = true;
           return true;
         } else {
-          continue;
+          flag = true;
         }
       }
 
       // try to get next batch data from next reader
-      while (usedIntervalFileIndex < sealedTsFiles.size()) {
+      while (!flag && usedIntervalFileIndex < sealedTsFiles.size()) {
         // init until reach a satisfied reader
         if (seriesReader == null || !seriesReader.hasNextBatch()) {
           IntervalFileNode fileNode = sealedTsFiles.get(usedIntervalFileIndex++);
           if (singleTsFileSatisfied(fileNode)) {
             initSingleTsFileReader(fileNode);
           } else {
-            continue;
+            flag = true;
```

Comment:
This code is no need!

Reply:
It's indeed a question

Reply:
A more delicate refactoring may be performed.